webapp/src/ts/services/telemetry.service.ts

```diff
@@ -164,65 +116,117 @@ export class TelemetryService {
+   */
   private storeConflictedAggregate(aggregateDoc) {
     aggregateDoc.metadata.conflicted = true;
-    aggregateDoc._id = [aggregateDoc._id, 'conflicted', Date.now()].join('-');
+    aggregateDoc._id = [ aggregateDoc._id, 'conflicted', Date.now() ].join(this.NAME_DIVIDER);
 
     return this.dbService
       .get({meta: true})
       .put(aggregateDoc);
   }
 
-  private aggregate(db) {
-    const reduceQuery = db.query(
-      {
-        map: (doc, emit) => emit(doc.key, doc.value),
-        reduce: '_stats',
-      },
-      {
-        group: true,
-      }
+  private getDBDate(dbName) {
+    const parts = dbName.split(this.NAME_DIVIDER);
+    return {
+      year: Number(parts[1]),
+      month: Number(parts[2]),
+      date: Number(parts[3]),
+    };
+  }
+
+  private async aggregate(db, dbName) {
+    const metadata = await this.generateMetadataSection(dbName);
+    const dbInfo = await this.dbService.get().info();
+    const reduceResult = await db.query(
+      { reduce: '_stats', map: (doc, emit) => emit(doc.key, doc.value) },
+      { group: true }
     );
 
-    return Promise
-      .all([
-        reduceQuery,
-        this.dbService.get().info(),
-        this.generateMetadataSection()
-      ])
-      .then(qAll => {
-        const reduceResult = qAll[0];
-        const infoResult = qAll[1];
-        const metadata = qAll[2];
-
-        const aggregateDoc: any = { type: 'telemetry' };
-
-        aggregateDoc.metrics = this.convertReduceToKeyValues(reduceResult);
-        aggregateDoc.metadata = metadata;
-        aggregateDoc._id = this.generateAggregateDocId(aggregateDoc.metadata);
-        aggregateDoc.device = this.generateDeviceStats();
-        aggregateDoc.dbInfo = infoResult;
-
-        return this.dbService
-          .get({ meta: true })
-          .put(aggregateDoc)
-          .catch(err => {
-            if (err.status === 409) {
-              return this.storeConflictedAggregate(aggregateDoc);
-            }
-
-            throw err;
-          });
-      });
+    const aggregateDoc = {
+      _id: this.generateAggregateDocId(metadata),
+      type: this.TELEMETRY_PREFIX,
+      metrics: this.convertReduceToKeyValues(reduceResult),
+      device: this.generateDeviceStats(),
+      metadata,
+      dbInfo,
+    };
+
+    try {
+      await this.dbService
+        .get({ meta: true })
+        .put(aggregateDoc);
+    } catch (error) {
+      if (error.status === 409) {
+        return this.storeConflictedAggregate(aggregateDoc);
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Moment when the aggregation starts (i.e. the beginning of the current day)
+   */
+  private getToday(): TodayMoment {
+    const today = moment().startOf('day');
+    return {
+      today,
+      formatted: today.format(this.DATE_FORMAT),
+    };
+  }
+
+  private async getCurrentTelemetryDB(today: TodayMoment) {
+    const telemetryDBs = await this.getTelemetryDBs();
```

Comment:
Minor, but since we already have this value in context in the `_record` function, it seems like we could take this as a function parameter and not load it again....